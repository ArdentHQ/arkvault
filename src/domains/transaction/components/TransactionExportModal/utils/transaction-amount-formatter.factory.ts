import { BigNumber } from "@ardenthq/sdk-helpers";
import { DTO, Helpers } from "@ardenthq/sdk-profiles";

const formatAmount = (amount: number, transaction: DTO.ExtendedConfirmedTransactionData) =>
	Helpers.Currency.format(amount, transaction.wallet().exchangeCurrency(), {
		withTicker: false,
	});

const multiPaymentAmount = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	if (transaction.isReceived()) {
		let totalReceived = BigNumber.make(transaction.amount());

		for (const recipient of transaction.recipients()) {
			if (recipient.address === transaction.wallet().address()) {
				totalReceived.minus(recipient.amount);
			}
		}

		return formatAmount(totalReceived.toNumber(), transaction);
	}

	return formatAmount(transaction.amount(), transaction);
};

const transactionAmount = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	if (transaction.isMultiPayment()) {
		return multiPaymentAmount(transaction);
	}

	return formatAmount(transaction.amount(), transaction);
};

const transactionTotal = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	if (transaction.isMultiPayment()) {
		return BigNumber.make(multiPaymentAmount(transaction)).plus(transaction.fee());
	}

	return formatAmount(transaction.total(), transaction);
};

export const AmountFormatter = (transaction: DTO.ExtendedConfirmedTransactionData) => ({
	amount: () => transactionAmount(transaction),
	fee: () => formatAmount(transaction.fee(), transaction),
	total: () => transactionTotal(transaction),
});
