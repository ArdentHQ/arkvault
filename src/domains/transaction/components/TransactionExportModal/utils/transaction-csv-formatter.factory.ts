import { BigNumber } from "@ardenthq/sdk-helpers";
import { DTO } from "@ardenthq/sdk-profiles";
import { CURRENCIES } from "@ardenthq/sdk-intl";
import { buildTranslations } from "@/app/i18n/helpers";

const recipient = (transaction: DTO.ExtendedConfirmedTransactionData) => {
	const { COMMON } = buildTranslations();

	if (transaction.isMultiPayment()) {
		return COMMON.MULTIPLE;
	}

	if (transaction.isTransfer()) {
		return transaction.recipient();
	}

	if (transaction.isVote() || transaction.isUnvote()) {
		return `${COMMON.VOTE} ${COMMON.TRANSACTION}`;
	}

	return COMMON.OTHER;
};

const multiPaymentAmount = (transaction: DTO.ExtendedConfirmedTransactionData): number => {
	if (transaction.sender() !== transaction.wallet().address()) {
		let totalReceived = BigNumber.ZERO;

		for (const recipient of transaction.recipients()) {
			if (recipient.address === transaction.wallet().address()) {
				totalReceived = totalReceived.plus(recipient.amount);
			}
		}

		return totalReceived.toNumber();
	}

	return transaction.amount();
};

const transactionAmount = (transaction: DTO.ExtendedConfirmedTransactionData): number => {
	const amount = transaction.isMultiPayment() ? multiPaymentAmount(transaction) : transaction.amount();

	if (transaction.isSent()) {
		return BigNumber.make(amount).times(-1).toNumber();
	}

	return amount;
};

const transactionFee = (transaction: DTO.ExtendedConfirmedTransactionData): number => {
	if (transaction.isSent()) {
		return BigNumber.make(transaction.fee()).times(-1).toNumber();
	}

	return 0;
};

const transactionTotal = (transaction: DTO.ExtendedConfirmedTransactionData): number => {
	if (transaction.isSent()) {
		return BigNumber.make(transaction.total()).times(-1).toNumber();
	}

	return transactionAmount(transaction);
};

const converted = (value: number, rate: number) => BigNumber.make(value).times(rate).toNumber();

const truncate = (value: number, currency: string) => {
	const decimals = CURRENCIES[currency]?.decimals ?? 8;

	return Math.trunc(value * 10 ** decimals) / 10 ** decimals;
};

export const CsvFormatter = (transaction: DTO.ExtendedConfirmedTransactionData, timeFormat: string) => {
	const amount = transactionAmount(transaction);
	const fee = transactionFee(transaction);
	const total = transactionTotal(transaction);

	const currency = transaction.wallet().currency();
	const exchangeCurrency = transaction.wallet().exchangeCurrency();

	const rate =
		transaction.total() === 0
			? 0
			: truncate(
					BigNumber.make(transaction.convertedTotal()).divide(transaction.total()).toNumber(),
					exchangeCurrency,
			  );

	return {
		amount: () => truncate(amount, currency),
		convertedAmount: () => truncate(converted(amount, rate), exchangeCurrency),
		convertedFee: () => (fee === 0 ? 0 : truncate(converted(fee, rate), exchangeCurrency)),
		convertedTotal: () => truncate(converted(total, rate), exchangeCurrency),
		datetime: () => transaction.timestamp()?.format(`DD.MM.YYYY ${timeFormat}`),
		fee: () => fee,
		rate: () => rate,
		recipient: () => recipient(transaction),
		sender: () => transaction.sender(),
		timestamp: () => transaction.timestamp()?.toUNIX(),
		total: () => truncate(total, currency),
	};
};
