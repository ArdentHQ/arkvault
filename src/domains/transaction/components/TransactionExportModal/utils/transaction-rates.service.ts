import { Contracts } from "@ardenthq/sdk-profiles";
import { DateTime } from "@ardenthq/sdk-intl";
import { BigNumber } from "@ardenthq/sdk-helpers";
import { HttpClient } from "@/app/services/HttpClient";

export const TransactionRates = ({ wallet }: { wallet: Contracts.IReadWriteWallet }) => {
	const exchangeEndpoint = "https://min-api.cryptocompare.com/data/histoday";
	const client = new HttpClient(0);
	const baseCurrency = "USD";

	const rates = {
		[baseCurrency]: new Map(),
	};

	const sync = async ({ from = DateTime.make(), to }: { from?: DateTime; to: DateTime }) => {
		const response = await client.get(exchangeEndpoint, {
			fsym: wallet.network().ticker(),
			limit: from.diffInDays(to),
			toTs: to.toUNIX(),
			tsym: baseCurrency,
		});

		for (const price of response.json().Data) {
			const ymd = DateTime.fromUnix(price.time).format("YYYY-MM-DD");
			rates[baseCurrency].set(ymd, BigNumber.make(price.close).toString());
		}
	};

	const byTimestamp = (timestamp?: DateTime) =>
		rates[baseCurrency].get(DateTime.make(timestamp).format("YYYY-MM-DD"));

	return {
		rate: (timestamp?: DateTime) => byTimestamp(timestamp),
		sync,
	};
};
