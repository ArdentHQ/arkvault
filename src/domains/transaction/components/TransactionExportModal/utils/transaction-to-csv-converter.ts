import { DTO } from "@ardenthq/sdk-profiles";
import { CsvSettings } from "@/domains/transaction/components/TransactionExportModal";
import { buildTranslations } from "@/app/i18n/helpers";
import { AmountFormatter } from "./transaction-amount-formatter.factory";

const headers = (settings: CsvSettings) => {
	const { COMMON } = buildTranslations();

	return [
		...(settings.includeTransactionId ? [COMMON.ID] : []),
		...(settings.includeDate ? [COMMON.TIMESTAMP, COMMON.DATE] : []),
		...(settings.includeSenderRecipient ? [COMMON.SENDER, COMMON.RECIPIENT] : []),
		...(settings.includeCryptoAmount ? [COMMON.AMOUNT, COMMON.FEE, COMMON.TOTAL] : []),
	].join(settings.delimiter);
};

const transactionToCsv = (transaction: DTO.ExtendedConfirmedTransactionData, settings: CsvSettings) => {
	const dateTimeFormat = "DD.MM.YYYY h:mm A";
	const timestamp = transaction.timestamp()?.toUNIX();
	const datetime = transaction.timestamp()?.format(dateTimeFormat);
	const sender = transaction.sender();
	const recipient = transaction.recipient();
	const amounts = AmountFormatter(transaction);

	const columns = [
		...(settings.includeTransactionId ? [transaction.id()] : []),
		...(settings.includeDate ? [timestamp, datetime] : []),
		...(settings.includeSenderRecipient ? [sender, recipient] : []),
		...(settings.includeCryptoAmount ? [amounts.amount(), amounts.fee(), amounts.total()] : []),
	];

	return columns.join(settings.delimiter);
};

export const convertToCsv = (transactions: DTO.ExtendedConfirmedTransactionData[], settings: CsvSettings) => {
	const rows = transactions.map((transaction) => transactionToCsv(transaction, settings));

	if (!settings.includeHeaderRow) {
		return rows.join("\n");
	}

	return [...[headers(settings)], ...rows].join("\n");
};
