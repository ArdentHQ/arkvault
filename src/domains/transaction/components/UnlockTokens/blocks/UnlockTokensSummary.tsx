import { DTO } from "@payvo/sdk-profiles";
import React, { useMemo } from "react";
import { useTranslation } from "react-i18next";

import {
	TransactionAmount,
	TransactionFee,
	TransactionTimestamp,
} from "@/domains/transaction/components/TransactionDetail";
import { TransactionSuccessful } from "@/domains/transaction/components/TransactionSuccessful";

interface UnlockTokensSummaryProperties {
	transaction: DTO.ExtendedSignedTransactionData;
}

export const UnlockTokensSummary = ({ transaction }: UnlockTokensSummaryProperties) => {
	const { t } = useTranslation();

	const wallet = useMemo(() => transaction.wallet(), [transaction]);

	return (
		<TransactionSuccessful
			transaction={transaction}
			senderWallet={wallet}
			title={t("TRANSACTION.UNLOCK_TOKENS.SUMMARY.TITLE")}
			description={t("TRANSACTION.UNLOCK_TOKENS.SUMMARY.DESCRIPTION")}
		>
			<TransactionAmount
				isTotalAmount
				amount={transaction.amount()}
				convertedAmount={transaction.convertedAmount()}
				currency={wallet.currency()}
				exchangeCurrency={wallet.exchangeCurrency()}
				isSent={false}
			/>

			<TransactionFee
				value={transaction.fee()}
				convertedValue={transaction.convertedFee()}
				currency={wallet.currency()}
				exchangeCurrency={wallet.exchangeCurrency()}
			/>

			<TransactionTimestamp timestamp={transaction.timestamp()} />
		</TransactionSuccessful>
	);
};
