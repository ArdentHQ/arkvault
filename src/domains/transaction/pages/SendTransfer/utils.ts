export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
