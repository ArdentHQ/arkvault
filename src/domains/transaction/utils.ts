import { Services } from "@ardenthq/sdk";
import { RecipientItem } from "@/domains/transaction/components/RecipientList/RecipientList.contracts";

export const isNoDeviceError = (error: any) => {
	if (!error) {
		return false;
	}

	if (String(error).includes("no device found")) {
		return true;
	}

	return false;
};

export const isRejectionError = (error: any) => String(error).includes("Condition of use not satisfied");

export const handleBroadcastError = ({ errors }: Services.BroadcastResponse) => {
	const allErrors = Object.values(errors);

	if (allErrors.length === 0) {
		return;
	}

	throw new Error(allErrors[0]);
};

export const getTransferType = ({ recipients }: { recipients: RecipientItem[] }): "multiPayment" | "transfer" => {
	const isMultiPayment = recipients.length > 1;
	return isMultiPayment ? "multiPayment" : "transfer";
};

export const withAbortPromise =
	(signal?: AbortSignal, callback?: () => void) =>
	<T>(promise: Promise<T>) =>
		new Promise<T>((resolve, reject) => {
			if (signal) {
				signal.addEventListener("abort", () => {
					callback?.();
					reject("ERR_ABORT");
				});
			}

			return promise.then(resolve).catch(reject);
		});
