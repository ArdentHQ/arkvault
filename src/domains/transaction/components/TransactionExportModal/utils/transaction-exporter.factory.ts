import { Contracts, DTO } from "@ardenthq/sdk-profiles";
import { Services } from "@ardenthq/sdk";
import { BigNumber } from "@ardenthq/sdk-helpers";
import { useState } from "react";
import { convertToCsv } from "./transaction-to-csv-converter";
import { CsvSettings } from "@/domains/transaction/components/TransactionExportModal";
import { assertString } from "@/utils/assertions";

interface TransactionExporterFetchProperties {
	type: "all" | "received" | "sent";
	dateRange?: Services.RangeCriteria;
	cursor?: number;
}

const filterTransactions = (transactions: DTO.ExtendedConfirmedTransactionData[]) =>
	transactions.filter((transaction) => {
		if (transaction.isTransfer()) {
			return transaction.sender() !== transaction.recipient();
		}

		if (transaction.isMultiPayment()) {
			let amount = BigNumber.make(transaction.amount());

			for (const recipient of transaction.recipients()) {
				if (transaction.sender() === recipient.address) {
					amount = amount.minus(recipient.amount);
				}
			}

			return !amount.isZero();
		}

		return false;
	});

export const TransactionExporter = ({
	profile,
	wallet,
	limit = 100,
}: {
	profile: Contracts.IProfile;
	wallet: Contracts.IReadWriteWallet;
	limit?: number;
}) => {
	const exchangeCurrency = profile.settings().get<string>(Contracts.ProfileSetting.ExchangeCurrency);
	const timeFormat = profile.settings().get<string>(Contracts.ProfileSetting.TimeFormat);

	let count = 0;

	assertString(exchangeCurrency);
	assertString(timeFormat);

	let requestedSyncAbort = false;

	let transactions: DTO.ExtendedConfirmedTransactionData[] = [];

	const sync = async ({
		type = "all",
		dateRange,
		cursor = 1,
	}: TransactionExporterFetchProperties): Promise<number | undefined> => {
		// Clear cache.
		if (cursor === 1) {
			transactions = [];
			requestedSyncAbort = false;
		}

		if (requestedSyncAbort) {
			return;
		}

		const page = await wallet.transactionIndex()[type]({ cursor, limit, timestamp: dateRange });

		transactions.push(...page.items());
		cursor = cursor + 1;

		count = transactions.length;

		// Last page.
		// TODO: Not relying on totalCount because it is an estimate
		//        and is not giving accurate pagination info. Address this issue after initial implementation.
		if (page.items().length < limit) {
			if (type === "received") {
				transactions = filterTransactions(transactions);
			}

			return transactions.length;
		}

		return sync({ cursor, dateRange, type });
	};

	return {
		transactions: () => ({
			abortSync: () => (requestedSyncAbort = true),
			count: () => count,
			items: () => transactions,
			sync,
			toCsv: (settings: CsvSettings) => convertToCsv(transactions, settings, exchangeCurrency, timeFormat),
		}),
	};
};
