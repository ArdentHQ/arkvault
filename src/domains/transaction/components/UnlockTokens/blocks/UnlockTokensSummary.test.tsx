import React from "react";
import { Route } from "react-router-dom";

import { TransactionFixture } from "@/tests/fixtures/transactions";
import { getDefaultProfileId, render } from "@/utils/testing-library";

import { UnlockTokensSummary } from "./UnlockTokensSummary";

const fixtureProfileId = getDefaultProfileId();

describe("UnlockTokensSummary", () => {
	it("should render", () => {
		const { asFragment } = render(
			<Route path="/profiles/:profileId">
				<UnlockTokensSummary
					transaction={{
						...TransactionFixture,
						wallet: () => ({
							...TransactionFixture.wallet(),
							currency: () => "ARK",
							id: () => "wallet-id",
							network: () => ({
								coin: () => "ARK",
								displayName: () => "ARK Devnet",
								id: () => "ark.devnet",
								isLive: () => false,
								isTest: () => true,
								ticker: () => "DARK",
							}),
						}),
					}}
				/>
			</Route>,
			{
				route: `/profiles/${fixtureProfileId}`,
			},
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
