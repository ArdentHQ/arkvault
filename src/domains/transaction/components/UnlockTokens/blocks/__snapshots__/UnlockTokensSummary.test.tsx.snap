// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnlockTokensSummary should render 1`] = `
<DocumentFragment>
  <section
    class="space-y-8"
    data-testid="TransactionSuccessful"
  >
    <div
      class="flex items-end justify-between bg-theme-background"
    >
      <div
        class="space-y-4"
      >
        <h1
          class="mb-0 text-2xl"
          data-testid="header__title"
        >
          Successfully Unlocked Balance
        </h1>
      </div>
    </div>
    <svg
      class="hidden w-full md:block"
    >
      transaction-success-banner-light-green.svg
    </svg>
    <p
      class="hidden text-theme-secondary-text md:block"
    >
      Your balance unlock transaction has been sent successfully.
    </p>
    <div
      class="flex flex-col overflow-hidden rounded-xl md:hidden"
    >
      <div
        class="css-1dwg83m"
      >
        <div
          class="css-v0ob3f"
          height="16"
          width="16"
        >
          <svg>
            circle-check-mark.svg
          </svg>
        </div>
        <span>
          Success
        </span>
      </div>
      <div
        class="css-k5jr4t"
      >
        Your balance unlock transaction has been sent successfully.
      </div>
    </div>
    <div>
      <div
        class="no-ligatures css-9iovq0"
        data-testid="TransactionDetail"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            ID
          </div>
          <div
            class="flex items-center font-semibold"
          >
            <div
              class="flex w-full items-center space-x-3 overflow-hidden"
            >
              <span
                class="overflow-hidden"
              >
                <a
                  class="ring-focus group inline-block flex items-center css-15tcy38"
                  data-ring-focus-margin="-m-1"
                  data-testid="Link"
                  href="https://live.arkscan.io/transaction/ee4175091d9f4dacf5fed213711c3e0e4cc371e37afa7bce0429d09bcf3ecefe"
                  rel="noopener noreferrer"
                >
                  <span
                    class="css-1y2194z"
                  >
                    <span
                      class="css-0"
                      data-testid="TruncateMiddle"
                    >
                      ea63bf9a4b3eaf75a1dfff7219â€¦7facf1aef29461868681b5c79b
                    </span>
                  </span>
                  <div
                    class="mb-[3px] shrink-0 align-middle duration-200 inline-block text-sm css-v0ob3f"
                    data-testid="Link__external"
                    height="16"
                    width="16"
                  >
                    <svg>
                      arrow-external.svg
                    </svg>
                  </div>
                </a>
              </span>
              <span
                class="flex text-theme-primary-300 dark:text-theme-secondary-600"
              >
                <button
                  class="ring-focus relative focus:outline-none"
                  data-ring-focus-margin="-m-1"
                  data-testid="clipboard-icon__wrapper"
                  type="button"
                >
                  <div
                    class="css-v0ob3f"
                    height="16"
                    width="16"
                  >
                    <svg>
                      copy.svg
                    </svg>
                  </div>
                </button>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="no-ligatures css-19rivvh"
        data-testid="TransactionType"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            Transaction Type
          </div>
          <div
            class="flex items-center font-semibold"
          >
            Transfer
          </div>
        </div>
        <div
          class="border-theme-text text-theme-text dark:border-theme-secondary-600 dark:text-theme-secondary-600 ring-theme-background duration-100 css-14jhlj1"
        >
          <div
            class="css-1i4b0eq"
            height="20"
            width="20"
          >
            <svg>
              transfer.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="no-ligatures css-19rivvh"
        data-testid="TransactionNetwork"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            Cryptoasset
          </div>
          <div
            class="flex items-center font-semibold"
          >
            LISK Testnet
          </div>
        </div>
        <div
          aria-label="LISK Testnet"
          class="text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 ring-theme-background duration-100 css-14jhlj1"
          data-testid="NetworkIcon-LSK-lsk.testnet"
        >
          <div
            class="css-1i4b0eq"
            data-testid="NetworkIcon__icon"
            height="20"
            width="20"
          >
            <svg>
              lsk.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="no-ligatures css-19rivvh"
        data-testid="TransactionSender"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            Sender
          </div>
          <div
            class="flex items-center font-semibold"
          >
            <div
              class="w-0 flex-1 overflow-auto text-right md:text-left"
            >
              <div
                class="no-ligatures flex overflow-hidden whitespace-nowrap items-center space-x-2 w-full"
              >
                <span
                  class="min-w-0 overflow-hidden text-theme-text font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center -space-x-1"
        >
          <div
            class="shrink-0 css-8d0u1i"
            data-testid="Avatar"
          >
            <div
              class="inline-flex h-full w-full items-center justify-center overflow-hidden rounded-full align-middle"
            >
              <img
                alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="no-ligatures css-19rivvh"
        data-testid="TransactionAmount"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            Total Amount
          </div>
          <div
            class="flex items-center font-semibold"
          >
            <div
              class="css-1vb1527"
              color="success"
            >
              <div
                class="flex space-x-1 items-stretch"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="Amount"
                >
                  + 100 LSK
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="border-theme-success-300 text-theme-success-600 dark:border-theme-success-600 ring-theme-background duration-100 css-14jhlj1"
        >
          <div
            class="css-1i4b0eq"
            height="20"
            width="20"
          >
            <svg>
              received.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="no-ligatures css-19rivvh"
        data-testid="TransactionFee"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            Transaction Fee
          </div>
          <div
            class="flex items-center font-semibold"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              21 LSK
            </span>
          </div>
        </div>
      </div>
      <div
        class="no-ligatures css-19rivvh"
        data-testid="TransactionTimestamp"
      >
        <div
          class="w-40 flex-1 space-y-2 whitespace-nowrap"
          data-testid="TransactionDetail--desktop"
        >
          <div
            class="no-ligatures text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            Timestamp
          </div>
          <div
            class="flex items-center font-semibold"
          >
            31.07.2020 4:34 PM
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
